tensor constant with wrapped method

avikchaudhuri opened the pull request:
Summary:
Tensor constants can show up through wrapped methods, so that they may not always be found in constant attributes. They need to be fakified and their meta vals need to be found to create graph signatures nevertheless. Otherwise non-strict barfs.

Longer term maybe we should pull this fakification up in non-strict.

Test Plan: added test

Differential Revision: D64480272

tugsbayasgalan commented on torch/export/_trace.py:
```diff
@@ -706,7 +706,7 @@ def _maybe_fixup_gm_and_output_node_meta(old_gm, new_gm):
         + len(graph_signature.input_tokens)
     )
-    set_missing_meta_vals(gm, flat_fake_args, total_non_user_inputs, constant_attrs)
+    set_missing_meta_vals(gm, flat_fake_args, total_non_user_inputs)
```
Hmm i wonder if the more correct thing to do is to beef up our constant attr finding logic? I guess it might be impossible if it is getting pulled from a method...

avikchaudhuri replied:
Yeah I suspect so.

tugsbayasgalan commented on torch/export/_trace.py:
```diff
@@ -1612,7 +1612,7 @@ def set_missing_meta_vals(gm, flat_args, num_params_buffers, constant_attrs):
         if node.op == "get_attr" and "val" not in node.meta:
             val = _get_attr(gm, node.target)
-            if val in constant_attrs and isinstance(val, torch.Tensor):
+            if isinstance(val, torch.Tensor):
```
Can you update the docstring of this helper function because now we don't care if it is a constant or not anymore? Maybe we can check if the val is not in param/buffers at least?

avikchaudhuri replied:
Sure, I can update the docstring. This path is only taken when a meta val doesn't exist, so it boils down to asserting that we've already handled params/buffers.

avikchaudhuri committed:
tensor constant with wrapped method (#138091)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/138091

Tensor constants can show up through wrapped methods, so that they may not always be found in constant attributes. They need to be fakified and their meta vals need to be found to create graph signatures nevertheless. Otherwise non-strict barfs.

Longer term maybe we should pull this fakification up in non-strict.

Test Plan: added test

Differential Revision: D64480272

avikchaudhuri commented:
@pytorchbot merge -f "Landed internally"
